platform/javascript/audio_driver_javascript.cpp

```diff
@@ -152,14 +175,64 @@ void AudioDriverJavaScript::finish() {
 	/* clang-format off */
 	EM_ASM({
 		_audioDriver_audioContext = null;
+		_audioDriver_audioInput = null;
 		_audioDriver_scriptNode = null;
 	});
 	/* clang-format on */
-	memdelete_arr(internal_buffer);
-	internal_buffer = NULL;
+
+	if (internal_buffer) {
+		memdelete_arr(internal_buffer);
+		internal_buffer = NULL;
+	}
+}
+
+Error AudioDriverJavaScript::capture_start() {
+
+	input_buffer.resize(buffer_length * 2 * 8);
+	input_position = 0;
+	input_size = 0;
+
+	/* clang-format off */
+	EM_ASM({
+		function gotMediaInput(stream) {
+			_audioDriver_audioInput = _audioDriver_audioContext.createMediaStreamSource(stream);
+			_audioDriver_audioInput.connect(_audioDriver_scriptNode);
+		}
+
+		function gotMediaInputError(e) {
+			console.log(e);
+		}
+
+		if (navigator.mediaDevices.getUserMedia) {
+			navigator.mediaDevices.getUserMedia({"audio": true}).then(gotMediaInput, gotMediaInputError);
+		} else {
+			if (!navigator.getUserMedia)
+				navigator.getUserMedia = navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+			navigator.getUserMedia({"audio": true}, gotMediaInput, gotMediaInputError);
+		}
+	});
+	/* clang-format on */
+
+	return OK;
+}
+
+Error AudioDriverJavaScript::capture_stop() {
+
+	/* clang-format off */
+	EM_ASM({
+		_audioDriver_audioInput.disconnect();
```

Comment:
We should `stop()` the MediaStreamTrack here to release the mic

Reply:
Added.

Reply:
Just fixed it.